src/script/main.ts

```diff
@@ -1,11 +1,13 @@
 export class Main {
     public static viPath;
+    public static viModel;
 
     public static LoadFile(filePath: string) {
-        const currentState = history.state;
+        const currentState = top.history.state;
         const search = "?q=" + filePath;
         if (currentState !== null) {
             const bdDiv: HTMLElement = document.getElementById("BlockDiagram");
+            sessionStorage.setItem(currentState.filePath.split("#")[0], JSON.stringify(currentState.parsedContent));
```

Comment:
I think you should stringify Main.viObj here instead of parsedContent. The last time UpdateHistory() would have been called is when we first create the HTML DOM for it. That does not give you the most recent state of the parsedContent, right? Unless the history stores the object references instead of a copy in its stack. (I doubt it's storing by reference, since we are creating one more object (`stateObj`) out of parsedContent.)

Reply:
No the `Main.viModel` is not initialized and thus the value for that key will get set to null in the session storage. 
I am updating the history after I switch states of multiframe in Element.ts so we have the latest sourcemodel.

Reply:
If Main.viModel is not initialised, I doubt the history will have a non-null value. And I have a feeling you won't need to update history or the sessionStorage every time the MultiFrameStructure changes frames.

Reply:
There's a TODO added for this now.